File: MetodosOrdenamiento.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class MetodosOrdenamiento {

    // Algoritmo de la burbuja
    public static void bubbleSort(List<Integer> arr) {
        int n = arr.size();
        for (int i = 0; i < n; i++) {
            boolean intercambio = false;
            for (int j = 0; j < n - i - 1; j++) {
                if (arr.get(j) > arr.get(j + 1)) {
                    // Intercambio simple con variable temporal
                    int temp = arr.get(j);
                    arr.set(j, arr.get(j + 1));
                    arr.set(j + 1, temp);
                    intercambio = true;
                }
            }
            // Si no hubo intercambios, ya está ordenado
            if (!intercambio) {
                break;
            }
        }
    }

    // Genera los datos según el estado
    public static List<Integer> generarDatos(int tamano, String estado) {
        List<Integer> datos = new ArrayList<>();

        if (estado.equals("aleatorio")) {
            Random rand = new Random();
            for (int i = 0; i < tamano; i++) {
                datos.add(rand.nextInt(1000000) + 1); // Número entre 1 y 1000000
            }
        } else if (estado.equals("invertido")) {
            for (int i = 0; i < tamano; i++) {
                datos.add(tamano - i); // Llenar de mayor a menor
            }
        }

        return datos;
    }

    public static void main(String[] args) {
        int[] tamanos = {100, 1000, 10000, 100000};
        String[] estados = {"aleatorio", "invertido"};

        System.out.printf("%-10s | %-10s | %-20s%n", "Tamaño", "Estado", "Tiempo (segundos)");
        System.out.println("------------------------------------------");

        for (int tamano : tamanos) {
            for (String estado : estados) {

                List<Integer> datos = generarDatos(tamano, estado);

                // Medir tiempo en nanosegundos y convertir a segundos
                long inicio = System.nanoTime();
                bubbleSort(datos);
                long fin = System.nanoTime();

                double tiempo = (fin - inicio) / 1_000_000_000.0;

                System.out.printf("%-10d | %-10s | %-20.6f s%n", tamano, estado, tiempo);
            }
        }
    }
}

//compilar y correr en la bash ya estoy harto
// javac MetodosOrdenamiento.java
// java MetodosOrdenamiento

File: MetodosOrdenamiento.js
// Algoritmo de la burbuja
function bubbleSort(arr) {
    let n = arr.length;
    for (let i = 0; i < n; i++) {
        let intercambio = false;
        for (let j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                // Intercambio simple con variable temporal
                let temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                intercambio = true;
            }
        }
        // Si no hubo intercambios, ya está ordenado
        if (!intercambio) {
            break;
        }
    }
}

// Genera los datos según el estado
function generarDatos(tamano, estado) {
    let datos = [];

    if (estado === "aleatorio") {
        for (let i = 0; i < tamano; i++) {
            // Número aleatorio entre 1 y 1,000,000
            datos.push(Math.floor(Math.random() * 1000000) + 1);
        }
    } else if (estado === "invertido") {
        for (let i = 0; i < tamano; i++) {
            datos.push(tamano - i); // Llenar de mayor a menor
        }
    }

    return datos;
}

function main() {
    const tamanos = [100, 1000, 10000, 100000];
    const estados = ["aleatorio", "invertido"];

    console.log("Tamaño     | Estado     | Tiempo (segundos)");
    console.log("------------------------------------------");

    for (let tamano of tamanos) {
        for (let estado of estados) {
            let datos = generarDatos(tamano, estado);

            // Medir el tiempo exacto con performance.now()
            let inicio = performance.now();
            bubbleSort(datos);
            let fin = performance.now();

            // Convertir milisegundos a segundos
            let tiempo = (fin - inicio) / 1000;

            console.log(`${tamano.toString().padEnd(10)} | ${estado.padEnd(10)} | ${tiempo.toFixed(6)} s`);
        }
    }
}

main();

//Correr este fukin programa
// node MetodosOrdenamiento.js
